app/src/main/java/com/ddoczi/tasky/agenda/data/remote/sync/AgendaSyncImpl.kt

```diff
@@ -0,0 +1,51 @@
+package com.ddoczi.tasky.agenda.data.remote.sync
+
+import androidx.work.BackoffPolicy
+import androidx.work.Constraints
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.ExistingWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
+import com.ddoczi.tasky.agenda.data.remote.worker.SyncLocalWithRemoteWorker
+import com.ddoczi.tasky.agenda.data.remote.worker.SyncRemoteWithLocalWorker
+import com.ddoczi.tasky.agenda.domain.sync.AgendaSync
+import java.time.Duration
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+
+class AgendaSyncImpl(
+    private val workManager: WorkManager
+): AgendaSync {
+    override fun syncAgendaRemoteWithLocal(): UUID {
+        val syncWorker = OneTimeWorkRequestBuilder<SyncRemoteWithLocalWorker>()
+            .setConstraints(
+                Constraints.Builder().setRequiredNetworkType(NetworkType.CONNECTED).build()
+            )
+            .setBackoffCriteria(BackoffPolicy.EXPONENTIAL, Duration.ofMinutes(5))
+            .build()
+        workManager.beginUniqueWork(
+            "sync_worker_local_to_remote",
+            ExistingWorkPolicy.APPEND_OR_REPLACE,
+            syncWorker
+        ).enqueue()
+        return syncWorker.id
+    }
+
+    override fun syncAgendaLocalWithRemote(): UUID {
+        val syncWorker = PeriodicWorkRequestBuilder<SyncLocalWithRemoteWorker>(
+            15, TimeUnit.MINUTES,
+            5, TimeUnit.MINUTES
+        ).setConstraints(
+            Constraints.Builder().setRequiredNetworkType(NetworkType.CONNECTED).build()
+        ).setBackoffCriteria(BackoffPolicy.EXPONENTIAL, Duration.ofMinutes(5)).build()
+
+        workManager.enqueueUniquePeriodicWork(
+            "sync_worker_remote_to_local",
+            ExistingPeriodicWorkPolicy.KEEP,
+            syncWorker
+        )
+        return syncWorker.id
```

Comment:
Also, how do you ensure that the syncs happen in the right order? The local changes have to be pushed to remote before you pull remote and update local